packages/trimerge-sync/src/TrimergeClient.ts

```diff
@@ -303,6 +310,10 @@ export class TrimergeClient<
     metadata: CommitMetadata,
     presence?: Presence,
   ): Promise<string | undefined> {
+    if (this.isShutdown) {
+      throw new Error('attempting to update doc after shutdown');
+    }
```

Comment:
I wonder if it's worth adding a `invariant` dep for this kind of thing. I've been using that a lot on the client and it's been quite nice